lib/runner/matchers/tags.js

```diff
@@ -1,27 +1,44 @@
 const Logger = require('../../util/logger.js');
 
 class TagsMatcher {
-  constructor(settings) {
-    this.settings = settings;
+  /**
+   * @param {string|Array} tagStringOrArray
+   * @return {Array}
+   */
+  static convertTags(tagStringOrArray) {
+    if (Array.isArray(tagStringOrArray)) {
+      return tagStringOrArray;
+    }
+
+    if (typeof tagStringOrArray == 'string' && tagStringOrArray.length > 0) {
+      return tagStringOrArray.split(',');
+    }
+
+    return [];
   }
 
+  constructor(settings) {
+    this.includeTags = TagsMatcher.convertTags(settings.tag_filter);
+    this.excludeTags = TagsMatcher.convertTags(settings.skiptags);
+  }
+  
   checkModuleTags(testModule) {
     let moduleTags = testModule['@tags'] || testModule.tags;
     let match = true;
 
     if (!Array.isArray(moduleTags)) {
-      return this.settings.tag_filter === undefined && this.settings.skiptags !== undefined;
+      return this.includeTags.length === 0;
     }
 
-    if (this.settings.tag_filter || this.settings.skiptags) {
+    if (this.includeTags || this.excludeTags) {
```

Comment:
This check now is a bit weird, since both of these properties now will always evaluate to something non-empty (an array). I think it can be removed altogether.

We should also rename "convertTagsToString", since the name is suggesting that they array will be converted to string, how about "mapTagsToString"?

Reply:
Check removed.
The second point is good! The module tags and filter and skip tags are now treated the same.